network/src/peers/peers.rs

```diff
@@ -352,11 +351,20 @@ impl Node {
         };
 
         // Limit set size.
-        let peers = peers
+        let mut peers: Vec<SocketAddr> = peers
             .choose_multiple(&mut SmallRng::from_entropy(), crate::SHARED_PEER_COUNT)
```

Comment:
```suggestion
            .choose_multiple(&mut SmallRng::from_entropy(), crate::SHARED_PEER_COUNT - 1)
```

Reply:
to account for the extra sync provider

Reply:
Replace the first address in the list with the sync provider instead, since this is only really applicable to beacons.